public/app/plugins/panel/xychart/TooltipView.tsx

```diff
@@ -88,6 +89,10 @@ export const TooltipView = ({
     const colorFacet = colorFacetFieldName ? findField(frame, colorFacetFieldName) : undefined;
     const sizeFacet = sizeFacetFieldName ? findField(frame, sizeFacetFieldName) : undefined;
 
+    if (colorFacet) {
+      color = colorFacet.display!(colorFacet.values[rowIndex]).color!;
```

Comment:
a little nervous using `!` here, maybe we should wrap this with some null coalescing?

Reply:
@ryantxu when does `display` or `color` not exist? :thinking: